apiserver/apiserver_test.go

```diff
@@ -0,0 +1,327 @@
+package apiserver
```

Comment:
Agreed, I've moved all the error cases to tables but left the golden path because each test case isn't isolated so it should be explicit that they aren't. Also when I wrote them in tables I ended up doubling the size of the tests.